rust/altrios-core/src/train/set_speed_train_sim.rs

```diff
@@ -384,20 +391,32 @@ impl SetSpeedTrainSim {
     /// - inertia
     /// - acceleration
     /// (some of these aren't implemented yet)
-    pub fn solve_required_pwr(&mut self, dt: si::Time) {
+    pub fn solve_required_pwr(&mut self, dt: si::Time) -> anyhow::Result<()> {
+        let pwr_pos_max =
+            self.loco_con.state.pwr_out_max.min(si::Power::ZERO.max(
+                self.state.pwr_whl_out + self.loco_con.state.pwr_rate_out_max * self.state.dt,
+            ));
+        let pwr_neg_max = self.loco_con.state.pwr_dyn_brake_max.max(si::Power::ZERO);
+        ensure!(
+            pwr_pos_max >= si::Power::ZERO,
+            format_dbg!(pwr_pos_max >= si::Power::ZERO)
+        );
+
         self.state.pwr_res = self.state.res_net() * self.speed_trace.mean(self.state.i);
         self.state.pwr_accel = self.state.mass_adj / (2.0 * self.speed_trace.dt(self.state.i))
             * (self.speed_trace.speed[self.state.i].powi(typenum::P2::new())
                 - self.speed_trace.speed[self.state.i - 1].powi(typenum::P2::new()));
         self.state.dt = self.speed_trace.dt(self.state.i);
 
         self.state.pwr_whl_out = self.state.pwr_accel + self.state.pwr_res;
+        self.state.pwr_whl_out = self.state.pwr_whl_out.max(-pwr_neg_max).min(pwr_pos_max);
```

Comment:
@SWRIganderson , this line was needed to enable `set_speed_train_sim_demo.py` to work with the results generated in `speed_limit_train_sim_demo.py`.  Even still, they don't produce line-on-line results because the set_speed version does not have friction brakes.  The brake planning stuff can happen only if the train is not forced to follow a particular speed trace because of the fact that service brakes are not instantaneous.  However, in ALTRIOS, our service brakes are instantanous so maybe it would be possible to change things to make these line on line.  

I'd like your thoughts on this, and we could meet online to talk through some of the nuanced stuff.

Reply:
Note to self @calbaker: verify that friction brakes are actually being used in the speed_limit_train_sim because if they aren't then the hypothesis that the change in aux load is due to change in dynamic braking doesn't hold water.